jenkins/ci.suse.de/pipelines/openstack-ardana-update.Jenkinsfile

```diff
@@ -0,0 +1,72 @@
+/**
+ * The openstack-ardana-tempest Jenkins Pipeline
+ *
+ * This job updates a pre-deployed CLM cloud.
+ */
+
+pipeline {
+
+  options {
+    // skip the default checkout, because we want to use a custom path
+    skipDefaultCheckout()
+    timestamps()
+  }
+
+  agent {
+    node {
+      label reuse_node ? reuse_node : "cloud-ardana-ci"
+      customWorkspace "${JOB_NAME}-${BUILD_NUMBER}"
+    }
+  }
+
+  stages {
+    stage('Setup workspace') {
+      steps {
+        script {
+          if (ardana_env == '') {
+            error("Empty 'ardana_env' parameter value.")
+          }
+          currentBuild.displayName = "#${BUILD_NUMBER}: ${ardana_env}"
+          // Use a shared workspace folder for all jobs running on the same
+          // target 'ardana_env' cloud environment
+          env.SHARED_WORKSPACE = sh (
+            returnStdout: true,
+            script: 'echo "$(dirname $WORKSPACE)/shared/${ardana_env}"'
+          ).trim()
+          if (reuse_node == '') {
+            sh('''
+              rm -rf $SHARED_WORKSPACE
+              mkdir -p $SHARED_WORKSPACE
+
+              # archiveArtifacts and junit don't support absolute paths, so we have to to this instead
+              ln -s ${SHARED_WORKSPACE}/.artifacts ${WORKSPACE}
+
+              cd $SHARED_WORKSPACE
+              git clone $git_automation_repo --branch $git_automation_branch automation-git
+              source automation-git/scripts/jenkins/ardana/jenkins-helper.sh
+              ansible_playbook load-job-params.yml
+              ansible_playbook setup-ssh-access.yml -e @input.yml
+            ''')
+          }
+        }
+      }
+    }
+
+    stage('Update ardana') {
+      steps {
+        sh('''
+          cd $SHARED_WORKSPACE
+          source automation-git/scripts/jenkins/ardana/jenkins-helper.sh
+          ansible_playbook ardana-update.yml -e @input.yml -e cloudsource=$update_to_cloudsource
+        ''')
+      }
+    }
+  }
+
+  post {
+    always {
+      archiveArtifacts artifacts: ".artifacts/**/*", allowEmptyArchive: true
```

Comment:
Just a thought, but why not use the same approach as in `openstack-ardana-vcloud` and let the upstream job deal with artifacts, if one is present ?

Reply:
As we talked before, we are having issues with broken artifact links submitted to rocketchat and OS-Health, so I choose to lower the number of jobs/days stored and archive artifacts on both upstream and downstream jobs to see how it goes.